docs/architecture/adr-080-libp2p-implementation.md

```diff
@@ -0,0 +1,157 @@
+other form. 
+
+## Decisions
+
+### Feature Parity
+
+The goal of this project should be a p2p stack that has feature parity
+with the legacy stack. Everything that is possible with the p2p stack
+today should be possible in the future, eventually reducing the
+overhead of peer communication and complexity in the gossip/peer layer
+of the system. 
+
+In persuit of this goal, this project has two explicit *non-goals*:
+
+- Do not support message/wire compatibility with legacy stack.
+- Do not support dual-mode (legacy/libp2p) in a single node or network.
+
+There does not seem to be a compelling use case of having a release
+that contains both P2P stacks. While either of these goals could
+be theoretically possible, by eliminating both of these goals we can
+effectively reduce the scope of the project and increase our
+confidence.
+
+### Target Iterative Progress 
+
+Rather than produce an extensive design document (other than this
+record of decisions,) we will work iteratively in a topic 
+branch--`main/libp2p`--to prototype the new libp2p integration. This
+work will aim to: 
+
+- develop a working (by way of the e2e tests), implementation of the
+  p2p layer without requiring changes to the reactor implementations.
+- migrate the reactor tests after completing a prototype. 
+- address design questions as they come up, by amending this document
+  or in periodic conversations over patches. 
+- minimize or avoid any interface changes between the reactor
+  implementations and the peer layer. 
+
+### Use libp2p Idiomatically 
+
+
+
+### Minimize P2P Code
+
+One of the largest benefits of this project is that we should be able
+to delete the vast majority of the existing p2p code, which should
+reduce the maintenance burden within Tendermint.
+
+### Defer Transport Decisions
+
+There are many high level aspects of p2p systems such as handshakes between
+nodes, transport protocols, which are fully configurable in the
+context of libp2p. There is no need to make decisions about these at
+this time, our decisions. 
+
+We should not, however, (necessarily) aim to delegate these decisions
+to the users. While we should make it possible for users to pass a
+custom libp2p connection object (`host.Host`) with limited support, we
+should **not** aim to offer too many customizations or configurations
+of different libp2p fundamentals.
+
+### Use GossipSub for Gossip Communication 
+
+An unfortunate consequence of the current p2p implementation is that
+reactors (e.g. mempool, consensus, and evidence) implement their own
+gossip logic, and maintain their own tracking of the connected peer
+network. These workloads should eventually migrate to using gosipsub
+(e.g. setting `broadcats: true` and letting libp2p broadcast,)
+however, this requires removing all of the legacy code. In the short
+term, we should leave the legacy gossip implementation for the first
+phase of the project. 
+
+## Detailed Design
```

Comment:
When we have converged on a design we should return to this section and fill it out in greater detail. We also shouldn't assume that the reader has knowledge of how the legacy stack is. 

I frankly still don't have any idea of how the plumbing of the libp2p fundamentals works but I'm hoping that this document could eventually serve as an entry-point (alternatively so could the spec)

Reply:
I added a bullet point about expanding this as we go.